src/compiler.rs

```diff
@@ -75,14 +86,65 @@ impl Compiler {
               .join(Self::expand_tilde(&relative.cooked)?)
               .lexiclean();
 
-            if import.is_file() {
+            let has_glob = import.as_os_str().as_encoded_bytes().contains(&b'*');
+
+            if has_glob {
+              if !(unstable || ast_unstable) {
+                return Err(Error::UnstableFeature {
+                  unstable_feature: UnstableFeature::GlobImport,
+                });
+              }
+
+              let import_base_str =
+                import
+                  .as_os_str()
+                  .to_str()
+                  .ok_or_else(|| Error::ImportGlob {
```

Comment:
```suggestion
                  .ok_or_else(|| Error::ImportGlobUnicode {
```

I'd like to use distinct error types instead of having a string message field.